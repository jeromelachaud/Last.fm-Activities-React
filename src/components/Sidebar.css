/* Sidebar */

.Logo {
  width: 5rem;
}

.Sidebar {
  width: 100%;
  position: relative;
  padding-bottom: 3rem;
  text-align: center;
  font-weight: 800;
}

@media screen and (min-width: 990px) {
  .Sidebar {
    width: auto;
    position: fixed;
    right: 20px;
    text-align: right;
  }
}
