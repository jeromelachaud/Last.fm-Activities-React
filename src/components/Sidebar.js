import React from 'react';
import Menu from './Menu';
import styled from 'styled-components';
import Logo from './Logo';

const SidebarWrapper = styled.div `
  width: 100%;
  position: relative;
  padding-bottom: 3rem;
  text-align: center;

  @media screen and (min-width: 990px) {
    width: auto;
    position: fixed;
    right: 20px;
    text-align: right;
  }
`;


let Sidebar = React.createClass ({

  propTypes: {
    user: React.PropTypes.object.isRequired
  },

  render() {
    const {
      user
    } = this.props;

    return (
      <SidebarWrapper>
        <Logo />
        <Menu
          user={user} />
      </SidebarWrapper>
    );
  }
});

export default Sidebar;
