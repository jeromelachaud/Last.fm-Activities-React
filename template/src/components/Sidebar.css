/* Sidebar */

.Logo {
  width: 5rem;
}

.Sidebar {
  position: fixed;
  right: 20px;
  width: auto;
  text-align: right;
  background-color: grey;
  color: white;
}
