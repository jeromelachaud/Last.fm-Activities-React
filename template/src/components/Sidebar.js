import React from 'react';
import Menu from './Menu';
import './Sidebar.css';
import logo from './logo.svg';


let Sidebar = React.createClass ({

  render() {

    const {
      user
    } = this.props;


    return (
      <div
        className="Sidebar">
        <img src={logo} className="Logo" alt="logo" />
        <Menu
          user={user} />
      </div>
    );
  }
});

export default Sidebar;
