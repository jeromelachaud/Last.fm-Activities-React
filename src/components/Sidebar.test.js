import React from 'React';
import {
  shallow
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Sidebar from './Sidebar';

describe('Sidebar componemts', () => {
  it('should render as expected', () => {
    const sidebarComponent = shallow(<Sidebar />);
    expect(toJson(sidebarComponent)).toMatchSnapshot();
  });
});
