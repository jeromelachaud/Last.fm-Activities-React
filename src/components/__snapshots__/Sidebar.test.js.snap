exports[`Sidebar componemts should render as expected 1`] = `
<div
  className="Sidebar">
  <img
    alt="logo"
    className="Logo"
    src="logo.svg" />
  <Menu />
</div>
`;
